src/Transit/Internal/Network.hs

```diff
@@ -98,8 +98,11 @@ tcpListener :: Bool -> IO (Maybe Socket)
 tcpListener True = return Nothing
 tcpListener False = do
   let hints' = defaultHints { addrFlags = [AI_NUMERICHOST], addrSocketType = Stream }
-  addr:_ <- getAddrInfo (Just hints') (Just "0.0.0.0") (Just (show defaultPort))
-  sock' <- socket (addrFamily addr) (addrSocketType addr) (addrProtocol addr)
+  -- getAddrInfo gives us a list of addresses. With AI_ADDRCONFIG, we
+  -- get one for IPv4 and another for IPv6
+  addr:_ <- getAddrInfo (Just hints') Nothing (Just (show defaultPort))
+  -- Always use AF_INET6 so that the socket listens on both IPv4 and IPv6
+  sock' <- socket AF_INET6 (addrSocketType addr) (addrProtocol addr)
```

Comment:
Hmm.  I think it depends on local system configuration whether this actually binds both AF_INET and AF_INET6.  For example, Linux has /proc/sys/net/ipv6/bindv6only (https://man7.org/linux/man-pages/man7/ipv6.7.html).  Apparently this defaults to false nowadays ... so maybe this is fine?  At least on Linux.  I'm not sure about macOS and Windows though.

Another case to consider - what if the system doesn't have any local IPv6 addresses?  Does this still work to bind to IPv4?

This is probably something it would be good to have CI covering since otherwise the potentially diverging behavior of multiple platforms makes it really difficult to get consistent results.